andrepereiradasilva/joomla-cms
# New Association table

### Summary of Changes

New Association Table for managing associations.
Can be further improved if needed.
### Testing Instructions

Store associations.

``` php
$table = JTable::getInstance('Association');
$table->storeAll('com_content.item', array('en-GB' => 2, 'pt-PT' => 12, 'fr-PT' => 18));
```

Load associations.

``` php
$table = JTable::getInstance('Association');
$results = $table->loadAll(array('id' => 2, 'context' => 'com_content.item'), '#__content'));
print_r($results);
```

And also the usual JTable methods.

Load a particular association.

``` php
$table = JTable::getInstance('Association');
$table->load(array('id' => 2, 'context' => 'com_content.item'));
echo $table->id . '|' . $table->context . '|' . $table->key;
```

Update a particular association.

``` php
$table = JTable::getInstance('Association');
$table->load(array('id' => 2, 'context' => 'com_content.item'));
$table->key = 'testkey';

// Bind, check and store the data.
if (!$table->check() || !$table->store())
{
    return false;
}
echo $table->id . '|' . $table->context . '|' . $table->key;
```

Delete a particular association.

``` php
$table = JTable::getInstance('Association');
$table->load(array('id' => 2, 'context' => 'com_content.item'));

// Bind, check and store the data.
if (!$table->delete())
{
    return false;
}
```
### Documentation Changes Required

The ususal.
